File: apps/web/src/types/document.ts
export interface DocumentSummary {
  id: string;
  importCaseId?: string;
  shipmentId?: string;
  shipmentContainerId?: string;
  documentType: string;
  title: string;
  documentNumber?: string;
  documentDate?: string;
  expiryDate?: string;
  status: string;
  notes?: string;
  createdAtUtc: string;
  updatedAtUtc: string;
  createdByUserId: string;
  createdByUserName?: string;
  rowVersion: number;
  currentVersion?: DocumentVersion;
}

export interface DocumentVersion {
  id: string;
  documentId: string;
  versionNumber: number;
  originalFileName: string;
  storedObjectKey: string;
  contentType: string;
  fileExtension: string;
  fileSizeBytes: number;
  sha256Hash: string;
  storageStatus: string;
  isCurrent: boolean;
  status: string;
  uploadedAtUtc: string;
  uploadedByUserId: string;
  uploadedByUserName?: string;
}

export interface DocumentChecklist {
  scopeType: string;
  scopeId: string;
  totalRequiredCount: number;
  completedCount: number;
  missingCount: number;
  status: string; // Complete, Missing, Expired
  items: DocumentChecklistItem[];
}

export interface DocumentChecklistItem {
  documentType: string;
  description: string;
  isRequired: boolean;
  status: string; // Complete, Missing, Expired
  linkedDocumentId?: string;
  documentTitle?: string;
  documentNumber?: string;
  expiryDate?: string;
}

File: apps/api/src/ImportControlTower.Domain/Entities/Document.cs
using System;
using System.Collections.Generic;

namespace ImportControlTower.Domain.Entities;

public class Document
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid? ImportCaseId { get; set; }
    public ImportCase? ImportCase { get; set; }

    public Guid? ShipmentId { get; set; }
    public Shipment? Shipment { get; set; }

    public Guid? ShipmentContainerId { get; set; }
    public ShipmentContainer? ShipmentContainer { get; set; }

    public string DocumentType { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? DocumentNumber { get; set; }
    public DateTime? DocumentDate { get; set; }
    public DateTime? ExpiryDate { get; set; }
    public string Status { get; set; } = "Active";
    public string? Notes { get; set; }

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
    public Guid CreatedByUserId { get; set; }
    public ApplicationUser? CreatedByUser { get; set; }
    public Guid? UpdatedByUserId { get; set; }
    public ApplicationUser? UpdatedByUser { get; set; }

    public ICollection<DocumentVersion> Versions { get; set; } = new List<DocumentVersion>();
}
